src/safeds/data/tabular/containers/_table.py

```diff
@@ -2193,26 +2198,64 @@ def plot_histograms(self) -> Image:
         import pandas as pd
-        import seaborn as sns
 
-        col_wrap = min(self.number_of_columns, 3)
+        n_cols = min(3, len(self.column_names))
+        n_rows = (len(self.column_names) + n_cols - 1) // n_cols
 
-        data = pd.melt(self._data.map(lambda value: str(value)), value_vars=self.column_names)
-        grid = sns.FacetGrid(data=data, col="variable", col_wrap=col_wrap, sharex=False, sharey=False)
-        grid.map(sns.histplot, "value")
-        grid.set_xlabels("")
-        grid.set_ylabels("")
-        grid.set_titles("{col_name}")
-        for axes in grid.axes.flat:
-            axes.set_xticks(axes.get_xticks())
-            axes.set_xticklabels(axes.get_xticklabels(), rotation=45, horizontalalignment="right")
-        grid.tight_layout()
-        fig = grid.fig
+        if n_cols == 1 and n_rows == 1:
+            one_col = True
+            fig, axs = plt.subplots(1, 1, tight_layout=True, figsize=(3, 3))
+        else:
+            one_col = False
+            fig, axs = plt.subplots(n_rows, n_cols, tight_layout=True, figsize=(n_cols * 3, n_rows * 3))
+
+        col_names = self.column_names
+        for col, ax in zip(col_names, axs.flatten() if not one_col else [axs]):
+            np_col = np.array(self.get_column(col))
+            bins = len(pd.unique(np_col))
+
+            if bins > n_bins:
+                bins = n_bins
+
+            ax.set_title(col)
+            ax.set_xlabel("")
+            ax.set_ylabel("")
+
+            if self.get_column(col).all(lambda x: not isinstance(x, (str | bool))):
+                np_col = np_col[~np.isnan(np_col)]
+                min_val = np.min(np_col)
+                max_val = np.max(np_col)
+
+                if bins < len(pd.unique(np_col)):
+                    n, bin_edges = np.histogram(self.get_column(col), bins, range=(min_val, max_val))
+
+                    bars = np.array([])
+                    for i in range(len(n)):
+                        bars = np.append(bars, f'{round(bin_edges[i], 2)}-{round(bin_edges[i+1], 2)}')
+
+                    ax.bar(bars, n)
+                    ax.set_xticks(np.arange(len(n)), bars, rotation=45, horizontalalignment="right")
+                else:
+                    unique_values = np.unique(np_col)
+                    n = np.array([np.sum(np_col == value) for value in unique_values])
+                    unique_values_str = [str(i) for i in unique_values]
+                    ax.bar(unique_values_str, n)
+                    ax.set_xticks(np.arange(len(unique_values_str)), unique_values_str, rotation=45, horizontalalignment="right")
+            else:
+                np_col = np_col.astype(str)
+                unique_values = np.unique(np_col)
+                n = np.array([np.sum(np_col == value) for value in unique_values])
+                ax.bar(unique_values, n)
+                ax.set_xticks(np.arange(len(unique_values)), unique_values, rotation=45, horizontalalignment="right")
```

Comment:
Disregarding the different ways of converting values to strings (`astype` should be preferred), these blocks are identical. They should be merged for better maintainability. Example (with some context):

```
            if self.get_column(col).type.is_numeric():
                np_col = np_col[~np.isnan(np_col)]

                if bins < len(pd.unique(np_col)):
                    min_val = np.min(np_col)
                    max_val = np.max(np_col)
                    n, bin_edges = np.histogram(self.get_column(col), bins, range=(min_val, max_val))

                    bars = np.array([])
                    for i in range(len(n)):
                        bars = np.append(bars, f'{round(bin_edges[i], 2)}-{round(bin_edges[i+1], 2)}')

                    ax.bar(bars, n)
                    ax.set_xticks(np.arange(len(n)), bars, rotation=45, horizontalalignment="right")
                    continue

            np_col = np_col.astype(str)
            unique_values = np.unique(np_col)
            hist = np.array([np.sum(np_col == value) for value in unique_values])
            ax.bar(unique_values, hist)
            ax.set_xticks(np.arange(len(unique_values)), unique_values, rotation=45, horizontalalignment="right")
```